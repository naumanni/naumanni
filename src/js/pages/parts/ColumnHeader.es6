/* @flow */
import React from 'react'
import classNames from 'classnames'

import {IconFont} from 'src/pages/parts'


type Props = {
  canShowMenuContent: boolean,
  isPrivate: boolean,
  menuContent: React.Element<any>,
  title: React.Element<any>,
  onClickHeader: (e: SyntheticEvent) => void,
  onClickMenu: (e: SyntheticEvent) => void,
}

export default class ColumnHeader extends React.PureComponent {
  props: Props

  render() {
    const {
      canShowMenuContent, isPrivate, menuContent, title,
      onClickHeader, onClickMenu} = this.props

    return (
      <div>
        <header
          className={classNames(
            'column-header',
            {'column-header-private': isPrivate}
          )}
          onClick={onClickHeader}
        >
          {title}
          <div className="column-headerMenu">
            <button className="column-headerMenuButton" onClick={onClickMenu}>
              <IconFont iconName="cog" />
            </button>
          </div>
        </header>
        {canShowMenuContent && menuContent}
      </div>
    )
  }
}
