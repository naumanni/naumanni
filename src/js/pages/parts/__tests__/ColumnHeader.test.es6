import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import ColumnHeader from '../ColumnHeader'


const takeSnapshot = (component) => {
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
}

describe('ColumnHeader', () => {
  let defaultProps = {
    canShowMenuContent: false,
    isPrivate: false,
    menuContent: <div>menu</div>,
    title: <div>title</div>,
    onClickHeader: jest.fn(),
    onClickMenu: jest.fn(),
  }

  describe('snapshot', () => {
    it('canShowMenuContent == false, isPrivate == false', () => {
      takeSnapshot(<ColumnHeader {...defaultProps} />)
    })

    it('canShowMenuContent == true, isPrivate == true', () => {
      const props = {
        ...defaultProps,
        canShowMenuContent: true,
        isPrivate: true,
      }
      takeSnapshot(<ColumnHeader {...props} />)
    })
  })

  it('propagate header click event', () => {
    const onClickHeader = jest.fn()
    const props = {
      ...defaultProps,
      onClickHeader,
    }
    const component = <ColumnHeader {...props} />
    const wrapper = shallow(component)
    const header = wrapper.find('.column-header')
    header.simulate('click')
    expect(onClickHeader).toHaveBeenCalled()
  })

  it('propagate menu button click event', () => {
    const onClickMenu = jest.fn()
    const props = {
      ...defaultProps,
      onClickMenu,
    }
    const component = <ColumnHeader {...props} />
    const wrapper = shallow(component)
    const header = wrapper.find('.column-headerMenuButton')
    header.simulate('click')
    expect(onClickMenu).toHaveBeenCalled()
  })
})
