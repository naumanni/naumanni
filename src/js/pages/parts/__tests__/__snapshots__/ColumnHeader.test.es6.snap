// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnHeader snapshot canShowMenuContent == false, isPrivate == false 1`] = `
<div>
  <header
    className="column-header"
    onClick={[Function]}
  >
    <div>
      title
    </div>
    <div
      className="column-headerMenu"
    >
      <button
        className="column-headerMenuButton"
        onClick={[Function]}
      >
        <span
          className="icon icon-cog"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`ColumnHeader snapshot canShowMenuContent == true, isPrivate == true 1`] = `
<div>
  <header
    className="column-header column-header-private"
    onClick={[Function]}
  >
    <div>
      title
    </div>
    <div
      className="column-headerMenu"
    >
      <button
        className="column-headerMenuButton"
        onClick={[Function]}
      >
        <span
          className="icon icon-cog"
        />
      </button>
    </div>
  </header>
  <div>
    menu
  </div>
</div>
`;
